### DISCUSSION TITLE
How to get createError message by using $fetch?

### DISCUSSION THREAD

**Discussion by maicss** [nuxt/nuxt#25429] [Questions] [answered]
Response in chrome devtools network tab:
![image](https://github.com/nuxt/nuxt/assets/9784767/b9682162-5125-4325-a19d-a4bd3d10871e)

Code: `$fetch({//some config}).catch(error => console.error(error))` get:
![image](https://github.com/nuxt/nuxt/assets/9784767/29c4106f-eb28-4a7f-b0cb-b05186203033)

How to get message I use `createError` to tell frontend `Index build failed: blabla` instead of `[PUT] "/api/collections/indexes": 500 Internal Server Error`?

Thanks!

**Comment by xjccc** [comment]
it's showed base on your error data?
```
$fetch({//some config}).catch(error => throw new Error(error.message))
```

**Reply by maicss** [reply]
Could you clarify it? I can't get it.

**Reply by xjccc** [reply, marked as answer]
emm.. i try this

```
// /api/hello.ts
export default defineEventHandler(() => {
  throw createError({
    statusCode: 500,
    statusMessage: 'this is error'
  })
})
```
```
// index.vue
$fetch('/api/hello').catch(error => {
  console.log(error, 'error');
  
  throw new Error(error.message)
})
```
![image](https://github.com/nuxt/nuxt/assets/16478776/3b68245f-05f1-4698-9cc4-156f79513dd5)

**Reply by maicss** [reply]
Yes, you are right. Use `statusMessage` instead of `message`. But I remembered one day I read document it suggested use `message` instead of `statusMessage`. So I change all `statusMessage` to `message`.

Thank you.

**Reply by maicss** [reply]
And another question, do you know how to remove status code from error message? Use regexp to delete first three number?

**Comment by nadirvishun** [comment]
you kan do like this: 
```
try {
  await $fetch("/api/hello")
} catch (e: unknown) {
  if (e instanceof FetchError) {
    console.log(e.data.message)
  }
}
```

**Reply by ahocevar** [reply]
This should be the accepted answer. Thanks, @nadirvishun.

You may need to
```js
import { FetchError } from 'ofetch';
